File: typescipt/src/compiler/JavaScript/builtInFunctions/dayFunctionCall.ts
import {SingleArgumentFunctionCall} from "./singleArgumentFunctionCall";
import {DayFunction} from "../../../language/expressions/functions/dayFunction";

export class DayFunctionCall extends SingleArgumentFunctionCall {
  protected override className = "BuiltInDateFunctions";
  protected override methodName = "day";

  constructor(functionNode: DayFunction) {
    super(functionNode);
  }
}

File: dotnet/Lexy/Lexy.Compiler/Compiler/CSharp/BuiltInFunctions/DayFunctionCall.cs
using Lexy.Compiler.Language.Expressions.Functions;
using Lexy.RunTime;

namespace Lexy.Compiler.Compiler.CSharp.BuiltInFunctions;

internal class DayFunctionCall : SingleArgumentFunctionCall<DayFunction>
{
    protected override string ClassName => nameof(BuiltInDateFunctions);
    protected override string MethodName => nameof(BuiltInDateFunctions.Day);
}